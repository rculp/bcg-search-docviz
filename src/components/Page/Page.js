import React from 'react';
import PropTypes from 'prop-types';
import { Grid } from 'react-flexbox-grid';

import Nav from 'components/Nav/Nav';
import './Page.scss';

const Page = ({ id, children }) => (
  <div id={id} className="page">
    <header>
      <Nav />
    </header>
    <main>
      <Grid>
        {children}
      </Grid>
    </main>
  </div>
);

Page.propTypes = {
  id: PropTypes.string,
  children: PropTypes.node.isRequired
};

Page.defaultProps = {
  id: null
};

export default Page;
