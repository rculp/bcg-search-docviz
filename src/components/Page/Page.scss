@import 'src/global-styles/responsive';

.page {
  position: relative;
  height: 100%;

  header {
    position: fixed;
    top: 0;
    width: 100%;
    z-index: 1;

    height: 56px;
    @media (max-width: map-get($grid-breakpoints, sm)) {
      height: 48px;
    }
  }

  main {
    height: 100%;
    width: 100%;

    padding-top: 56px;
    @media (max-width: map-get($grid-breakpoints, sm)) {
      padding-top: 48px;
    }
  }
}
