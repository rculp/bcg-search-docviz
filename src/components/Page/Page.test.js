import React from 'react';
import { shallow } from 'enzyme';

import Page from './Page';

describe(Page, () => {
  let props;

  beforeEach(() => {
    props = {
      test: undefined
    };
  });

  it('renders and matches our snapshot', () => {
    const component = shallow(
      <Page>
        <div></div>
      </Page>
    );
    expect(component).toMatchSnapshot();
  });

  it('passes children', () => {
    const component = shallow(
      <Page {...props}>
        <div>Test</div>
      </Page>
    );
    expect(component.contains(<div>Test</div>)).toEqual(true);
  });
});