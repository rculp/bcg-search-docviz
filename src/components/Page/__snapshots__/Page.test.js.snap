// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page renders and matches our snapshot 1`] = `
<div
  className="page"
  id="test-page"
>
  <BcgHeader />
  <div />
</div>
`;
